apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore/Logging/GoogleLogger.cs

```diff
@@ -89,6 +94,17 @@ public sealed class GoogleLogger : ILogger
             _fullLogName = logTarget.GetFullLogName(_loggerOptions.LogName);
             _serviceProvider = serviceProvider;
             _clock = clock ?? SystemClock.Instance;
+            // Create the service context here, it's going to be the same for all entries.
+            if (_loggerOptions.ServiceName != null)
```

Comment:
Perhaps `LoggerOptions` should have a `CreateServiceContext` or `GetServiceContext` method? (That would help to simplify things if we allow complete customization too.)

Reply:
OK, I reshuffled the code a little bit. Squashed everything in the original commit so you can review as a whole.